### PR TITLE
Cleanups to `ivf_index()` C++ code, and small cleanups in Python

### PR DESCRIPTION
**Pull request by jparismorgan**
### What
While reading this code I made renames, added comments, and removed unused code to help my understanding. There should be no logic or behavior change here, just a no-op refactor.

### Testing
Tests pass.

### CODE REVIEW

File: src/include/detail/ivf/index.h [hunk 1]
```diff
@@ -5,7 +5,7 @@
  *
  * The MIT License
  *
- * @copyright Copyright (c) 2023 TileDB, Inc.
+ * @copyright Copyright (c) 2023 Tileinput_vectors, Inc.
```
Comment (cainamisir) [inline src/include/detail/ivf/index.h:8]: nit: I think you replaced this by mistake

**Review by cainamisir — APPROVED** [review]
looks good!

File: apis/python/src/tiledb/vector_search/ingestion.py [hunk 2]
```diff
@@ -1795,41 +1783,25 @@ def ingest_vectors_udf(
...
+                    index_array_uri=partial_write_array_index_uri,
+                    id_uri=partial_write_array_ids_uri,
+                    start=part,
+                    end=part_end,
+                    nthreads=threads,
+                    **(
+                        {"timestamp": index_timestamp}
+                        if index_timestamp is not None
+                        else {}
+                    ),
```
Comment (jparismorgan) [inline apis/python/src/tiledb/vector_search/ingestion.py:1802]: Just to share, beyond normal testing, I also tested this with a small example:
```
def foo(cool, timestamp = 1):
    print('timestamp', timestamp)

def test_foo():
    print()

    foo('a', timestamp=0)
    
    foo('a', 0)

    index_timestamp = None
    foo(
        'a', 
        **({"timestamp": index_timestamp} if index_timestamp is not None else {}
    )
    )

    index_timestamp = 2
    foo(
        'a', 
        **({"timestamp": index_timestamp} if index_timestamp is not None else {}
    )
    )
```
which prints:
```
timestamp 0
timestamp 0
timestamp 1
timestamp 2
```

File: src/include/detail/ivf/index.h [hunk 3]
```diff
@@ -95,51 +95,54 @@ int ivf_index(
...
     // debug_matrix(indices, "indices");
 
     // Some variables for debugging
     // @todo remove these once we are confident in the code
-    auto mis = std::max_element(begin(indices), end(indices));
-    auto a = std::distance(begin(indices), mis);
-    auto b = std::distance(mis, end(indices));
-    auto misx = *mis;
+    // auto mis = std::max_element(begin(indices), end(indices));
+    // auto a = std::distance(begin(indices), mis);
```
Comment (NikolaosPapailiou) [inline src/include/detail/ivf/index.h:135]: Why don't we remove these entirely?

**Review by NikolaosPapailiou — APPROVED** [review]

File: src/include/detail/ivf/index.h (hunk 3 above)
Comment (jparismorgan) [reply src/include/detail/ivf/index.h]: Sure, done, just was leaving b/c the comments somewhat suggested there was future work we wanted to do here with it.

**Pull request merged**